### PR TITLE
Update action.yml

### PR DESCRIPTION
**Pull request by balakrishna-deriv**

### CODE REVIEW

**Review by review-deriv — COMMENTED** [review]
PR description is too short. You can use this format:

```
# Description
This is a summary of the changes. This should describe why do we need this change.
Whys and Hows come here.

# Cards
* https://clickup.com/12345
* https://clickup.com/12346

```

File: .github/actions/checkout/action.yml [hunk 1]
```diff
@@ -43,13 +43,13 @@ runs:
               if git ls-remote --heads --quiet --exit-code https://${{ inputs.token }}@github.com/${{ inputs.repository }}.git ${{ inputs.ref }}
               then
                 echo "::notice::Checkout: ${{ inputs.repository }} using ${{ inputs.ref }}"
-                echo "ref-exists=true" >> $GITHUB_OUTPUT
+                echo "::set-output name=ref-exists::true"  # Use ::set-output for boolean output
```
Comment (balakrishna-deriv) [inline .github/actions/checkout/action.yml:46]:
```suggestion
                echo "::set-output name=ref-exists::true" 
```

File: .github/actions/checkout/action.yml [hunk 2]
```diff
@@ -43,13 +43,13 @@ runs:
...
               else
                 echo "::notice::Checkout: ${{ inputs.repository }} does not have ref ${{ inputs.ref }} (fallback to ${{ inputs.alternate_ref }})"
-                echo "ref-exists=false" >> $GITHUB_OUTPUT
+                echo "::set-output name=ref-exists::false"  # Use ::set-output for boolean output
```
Comment (balakrishna-deriv) [inline .github/actions/checkout/action.yml:49]:
```suggestion
                echo "::set-output name=ref-exists::false"
```

File: .github/actions/checkout/action.yml [hunk 3]
```diff
@@ -58,7 +58,7 @@ runs:
               ref: ${{ inputs.ref }}
               token: ${{ inputs.token }}
 
-        - if: steps.repo.outputs.ref-exists == 'false'
+        - if: !steps.repo.outputs.ref-exists  # Use ! for negation
```
Comment (balakrishna-deriv) [inline .github/actions/checkout/action.yml:61]:
```suggestion
        - if: !steps.repo.outputs.ref-exists
```

**Pull request closed without merging**